kimchi/js/bindings.js

```diff
@@ -2353,11 +2380,37 @@ var caml_pasta_fp_commitments_to_rust = function (x) {
   var w_comm = convertArray(x[1]);
   var z_comm = caml_vesta_poly_comm_to_rust(x[2]);
   var t_comm = caml_vesta_poly_comm_to_rust(x[3]);
-  return new plonk_wasm.WasmFpProverCommitments(w_comm, z_comm, t_comm);
+  var lookup = caml_opt_to_rust(x[4], caml_fp_lookup_commitments_to_rust);
+  return new plonk_wasm.WasmFpProverCommitments(w_comm, z_comm, t_comm, lookup);
 };
 
-// Provides: caml_pasta_fp_commitments_of_rust
+// Provides: caml_fp_lookup_commitments_of_rust
 // Requires: caml_vesta_poly_comm_of_rust, js_class_vector_of_rust_vector, plonk_wasm
+var caml_fp_lookup_commitments_of_rust = function (wasm_lc) {
+  var convertArray = function (v) {
+    var a = js_class_vector_of_rust_vector(v, plonk_wasm.WasmFpPolyComm);
+    var res = [0];
+    for (var i = 0; i < a.length; ++i) {
+      res.push(caml_vesta_poly_comm_of_rust(a[i]));
+    }
+    return res;
+  };
+
+  var sorted = convertArray(wasm_lc.sorted);
+  var aggreg = caml_vesta_poly_comm_of_rust(wasm_lc.aggreg);
+  var wasm_lc_runtime = wasm_lc.runtime;
+  var caml_runtime;
+  if (wasm_lc_runtime === undefined) {
+    caml_runtime = 0;
+  } else {
+    caml_runtime = [0, caml_vesta_poly_comm_of_rust(wasm_lc_runtime)];
+  }
```

Comment:
I think there might be a `opt_of_rust` function that does this as a one-liner.

Reply:
You're actually using it below.

Reply:
Yes, it seems that even in develop I haven't done it. I'm opening a follow-up issue to make both branch the same.